pkg/controllers/user/authz/podsecuritypolicy/binding.go

```diff
@@ -0,0 +1,140 @@
+func RegisterBindings(context *config.UserContext) {
+	logrus.Infof("registering podsecuritypolicy project handler for cluster %v", context.ClusterName)
+
+	lifecycle := &lifecycle{
+		policyLister:         context.Extensions.PodSecurityPolicies("").Controller().Lister(),
+		policies:             context.Extensions.PodSecurityPolicies(""),
+		psptLister:           context.Management.Management.PodSecurityPolicyTemplates("").Controller().Lister(),
+		clusterRoleLister:    context.RBAC.ClusterRoles("").Controller().Lister(),
+		clusterRoles:         context.RBAC.ClusterRoles(""),
+		serviceAccountLister: context.Core.ServiceAccounts("").Controller().Lister(),
+		serviceAccounts:      context.Core.ServiceAccounts("").Controller(),
+	}
+
+	context.Management.Management.PodSecurityPolicyTemplateProjectBindings("").
+		AddLifecycle("PodSecurityPolicyTemplateProjectBindingsLifecycleHandler", lifecycle)
+}
+
+type lifecycle struct {
+	policyLister         v1beta1.PodSecurityPolicyLister
+	policies             v1beta1.PodSecurityPolicyInterface
+	psptLister           v3.PodSecurityPolicyTemplateLister
+	clusterRoleLister    v12.ClusterRoleLister
+	clusterRoles         v12.ClusterRoleInterface
+	serviceAccountLister v1.ServiceAccountLister
+	serviceAccounts      v1.ServiceAccountController
+}
+
+func (l *lifecycle) Create(obj *v3.PodSecurityPolicyTemplateProjectBinding) (*v3.PodSecurityPolicyTemplateProjectBinding, error) {
+	return l.sync(obj)
+}
+
+func (l *lifecycle) Updated(obj *v3.PodSecurityPolicyTemplateProjectBinding) (*v3.PodSecurityPolicyTemplateProjectBinding, error) {
+	return l.sync(obj)
+}
+
+func (l *lifecycle) Remove(obj *v3.PodSecurityPolicyTemplateProjectBinding) (*v3.PodSecurityPolicyTemplateProjectBinding, error) {
+	return obj, resyncServiceAccounts(l.serviceAccountLister, l.serviceAccounts, "")
+}
+
+func (l *lifecycle) sync(obj *v3.PodSecurityPolicyTemplateProjectBinding) (*v3.PodSecurityPolicyTemplateProjectBinding, error) {
+	if obj.PodSecurityPolicyTemplateName == "" {
+		return obj, nil
+	}
+
+	podSecurityPolicyName := fmt.Sprintf("%v-psp", obj.PodSecurityPolicyTemplateName)
+	_, err := l.policyLister.Get("", podSecurityPolicyName)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			_, err = l.createPolicy(obj, podSecurityPolicyName)
+			if err != nil {
+				return nil, fmt.Errorf("error creating policy: %v", err)
+			}
+		} else {
+			return nil, fmt.Errorf("error getting policy: %v", err)
+		}
+	}
+
+	clusterRoleName := fmt.Sprintf("%v-clusterrole", obj.PodSecurityPolicyTemplateName)
+	_, err = l.clusterRoleLister.Get("", clusterRoleName)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			_, err = l.createClusterRole(clusterRoleName, podSecurityPolicyName, obj)
+			if err != nil {
+				return nil, fmt.Errorf("error creating cluster role: %v", err)
+			}
+		} else {
+			return nil, fmt.Errorf("error getting cluster role: %v", err)
+		}
+	}
+
+	return obj, resyncServiceAccounts(l.serviceAccountLister, l.serviceAccounts, "")
```

Comment:
Since a binding is for a particular project and a project is a subset of the cluster's namespaces, enqueueing _all_ service accounts in the cluster is overly aggressive. You should get all the namespaces in the project and then just enqueue the svc accounts for those namespaces. (You can actually just enqueue those namespaces themselves and they will in turn enquque their service accounts when the namespace handler you wrote gets executedO